proto/rpc/v1/transaction.proto

```diff
@@ -0,0 +1,57 @@
+syntax = "proto3";
+
+package rpc.v1;
+
+import "rpc/v1/amount.proto";
+
+// A class encompassing all transactions.
+message Transaction {
+    // The account originating the transaction.
+    string account = 1;
+
+    // The fee attached to the transaction.
+    XRPAmount fee = 2;
+
+    // The sequence number for the transaction.
+    uint32 sequence = 3;
+
+    // Data specific to a the type of transaction being submitted.
+    oneof transaction_data {
+        Payment payment = 4;
+    }
+
+    // The public key of the account which signed the transaction in hexadecimal.
+    bytes signing_public_key_hex = 5;
+
+    bytes signature = 6;
+
+    uint32 flags = 7;
+
+    uint32 last_ledger_sequence = 8;
+    
+    repeated Path paths = 9;
+
+    CurrencyAmount send_max = 10;
+}
+
+message Path {
+
+    repeated PathElement elements = 1;
+}
+
+message PathElement {
+
+    string account = 1;
+
+    string currency = 2;
+
+    string issuer = 3;
+}
+
+message Payment {
```

Comment:
I believe (and CJ can correct me) that this is at least the basics we would need to start building XRP transactions here. We just need to make sure what we have is extensible and doesn't cause breaking changes in the future.

Reply:
I'm just going to add the missing fields. SendMax and Paths will be moved from the Transaction type to here. I see that DestinationTag, InvoiceID and DeliverMin are missing. Any others?

Reply:
added fields, lmk if i missed any (non-deprecated) fields

Reply:
Are these fields getting added?

Reply:
The fields to be added were added. If there are fields missing, please point them out. Note though, we are only supporting Payment transactions at this time, so some fields necessary for other transaction types were not added.

Reply:
Fields missing; TransactionType, AccountTxnID. But everything else looks good. Transaction, TX and meta. You're returning all the necessary information via messages.

Reply:
TransactionType is omitted, since the type is defined via protobuf; each transaction type has it's own unique protobuf message type.
You are right that AccountTxnID is missing from the Transaction message. I added it to the AccountRoot ledger object message, but missed it in the Transaction message. I'll add that now; thanks for pointing it out.

Reply:
Oh cool. So I'm looking to add Escrow next. Seems I would follow the message format of the Payment protobuf? I can help out as my use case is wrapping up and Escrow is next.

Reply:
@dangell7 I added AccountTxnID (naming is a bit different though). 

As to adding Escrow: I assume you mean adding protobuf messages for EscrowCreate, EscrowFinish and EscrowCancel. You are free to do so and create a PR, and yes, follow the same format as the Payment message, but you are going to have to also implement that functionality on the C++ side as well, in addition to adding the protobuf messages. Also, you would need to also add support (in protobuf and C++) for all ledger objects that those Escrow transaction types affect, in order to return complete meta for those Escrow transactions.